File: 90. Subsets II/Java/Subsets II.java
public class Solution {
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        class Utils {
            void subsets(int[] nums, List<List<Integer>> result, List<Integer> current, int start) {
                result.add(new ArrayList<>(current));
                for (int i = start; i < nums.length; i++) {
                    if (i == start || nums[i] != nums[i - 1]) {
                        current.add(nums[i]);
                        subsets(nums, result, current, i + 1);
                        current.remove(current.size() - 1);
                    }
                }
            }
        }
        Arrays.sort(nums);
        List<List<Integer>> result = new ArrayList<>();
        Utils utils = new Utils();
        utils.subsets(nums, result, new ArrayList<>(), 0);
        return result;
    }
}

File: 90. Subsets II/Python/Subsets II.py
class Solution(object):
    def subsetsWithDup(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """

        def subsets(nums, result, current, start):
            result.append(current[:])
            for i in range(start, len(nums)):
                if i == start or nums[i] != nums[i - 1]:
                    current.append(nums[i])
                    subsets(nums, result, current, i + 1)
                    current.pop()

        nums.sort()
        result = []
        subsets(nums, result, [], 0)
        return result
